onnxruntime/core/framework/endian_utils.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cstring>
+#include <type_traits>
+
+#include "gsl/gsl"
+
+#include "core/common/status.h"
+#include "core/framework/endian.h"
+
+namespace onnxruntime {
+namespace utils {
+
+/**
+ * Swaps the byte order of elements in a buffer.
+ * This is a low-level funtion - please be sure to pass in valid arguments.
+ * In particular, source_bytes and destination_bytes should have the same size,
+ * which should be a multiple of element_size_in_bytes. element_size_in_bytes
+ * should also be greater than zero.
+ *
+ * @param element_size_in_bytes The size of an individual element, in bytes.
+ * @param source_bytes The source byte span.
+ * @param destination_bytes The destination byte span.
+ */
+void SwapByteOrderCopy(size_t element_size_in_bytes, gsl::span<const char> source_bytes, gsl::span<char> destination_bytes);
+
+/**
+ * Copies between two buffers where one is little-endian and the other has
+ * native endian-ness.
+ */
+template <size_t ElementSize>
```

Comment:
>size_t ElementSize [](start = 10, length = 18)

Consider making this template off class T instead of size_t. The reason is bc it creates a huge number of instantiations for the template. It is too easy to do sizeof(T) inside the function and have a lot less instances of the template.

Reply:
Good point about the multiple instantiations. I made it a function argument and moved this function to a detail namespace.